### PR TITLE
tests: enable unit testing via tox

### PR DESCRIPTION
**Pull request by smithfarm**
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -164,20 +164,53 @@ Remember to re-run `pip install --editable .` after each git pull.
...
+
+```
+zypper -n install python3-tox
+```
+
+Then, execute the following commands in the top-level of your local git clone
+to install the dependencies, including test dependencies:
+
+```
+pip install --editable ./[dev]
```
Comment (toabctl) [inline README.md]: is `/` needed here?

File: README.md [hunk 2]
```diff
@@ -164,20 +164,53 @@ Remember to re-run `pip install --editable .` after each git pull.
...
 ```
-zypper -n install python3-tox python3-pylint
+
+Finally, inspect the list of testing environments in `tox.ini` and choose one or
+more that you are interested in. Here is an example, but the actual output might
+be different:
+
 ```
+$ grep envlist tox.init
```
Comment (toabctl) [inline README.md]: This is `tox --listenv`

File: tests/test_pytest_smoke.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import pytest
```
Comment (toabctl) [inline tests/test_pytest_smoke.py]: can you remove `_pytest` from the filename? not needed.

File: tests/test_pytest_smoke.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+
+def test_pytest_smoke():
```
Comment (toabctl) [inline tests/test_pytest_smoke.py]: can you remove `_pytest` from the function name? not needed

File: setup.cfg [hunk 5]
```diff
@@ -48,8 +48,8 @@ console_scripts =
 [options.extras_require]
 dev =
     pytest
-    pycodestyle==2.5.0
-    pylint==2.4.4
+    pycodestyle
```
Comment (toabctl) [inline setup.cfg]: I like the commit message! :)

File: README.md (hunk 1 above)
Comment (smithfarm) [reply README.md]: Don't know. I tested this form of the command and it works.

File: tests/test_pytest_smoke.py (hunk 3 above)
Comment (smithfarm) [reply tests/test_pytest_smoke.py]: It's needed for me :-) This file contains a "pytest smoke test" - i.e. if the test passes, we know pytest is working.

File: tests/test_pytest_smoke.py (hunk 4 above)
Comment (smithfarm) [reply tests/test_pytest_smoke.py]: negative. see the other comment

File: README.md (hunk 2 above)
Comment (smithfarm) [reply README.md]: changing to `tox --listenvs`

File: tests/test_pytest_smoke.py (hunk 3 above)
Comment (toabctl) [reply tests/test_pytest_smoke.py]: yeah. but usually python test files are just called `test_....py`. Independent from the test framework and test runner we use.

Comment (smithfarm) [reply tests/test_pytest_smoke.py]: In this project, I prefer to be explicit that we are using `pytest`

Comment (smithfarm) [reply tests/test_pytest_smoke.py]: (note, I'm not suggesting that we name all the files `test_pytest_....py` :-) Just this one.

**Comment by smithfarm** [comment]
@toabctl Would `test_testing_subsystem_smoke` be less unacceptable for you?

File: tests/test_pytest_smoke.py (hunk 3 above)
Comment (toabctl) [reply tests/test_pytest_smoke.py]: ok :)

**Pull request merged**